src/core/common/message.hpp

```diff
@@ -1471,9 +1471,9 @@ class MessagePool : public InstanceLocator, private NonCopyable
 private:
     Buffer *NewBuffer(Message::Priority aPriority);
     void    FreeBuffers(Buffer *aBuffer);
-    otError ReclaimBuffers(int aNumBuffers, Message::Priority aPriority);
+    otError ReclaimBuffers(Message::Priority aPriority);
 
-#if OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT == 0
+#if OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT == 0 && OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE == 0
```

Comment:
I know this is existing code, but since we are changing this line here, how about changing it to use `!` instead of `== 0` ?

We more often use the `!OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT` pattern. I pesonally prefer it and feel it is more readable compared to `== 0` check.

Reply:
agree